### PR TITLE
feat: show symbol on approval

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

Show symbol on approval transactions

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

most closely related to loose ends from https://github.com/shapeshift/lib/issues/658 (already closed)

## Risk

- transaction history renders correctly

## Testing

- approval tx's should now show the symbol

## Screenshots (if applicable)

before

image

after

image

### CODE REVIEW

File: src/components/TransactionHistoryRows/TransactionContract.tsx [hunk 1]
```diff
@@ -24,24 +27,31 @@ export const TransactionContract = ({
...
     isReceive && !txDetails.tx.data?.method ? txDetails.tradeTx?.type : txDetails.tx.data?.method
   const isFirstAssetOutgoing = interactsWithWithdrawMethod && isSend
 
+  const titlePrefix = translate(
+    txDetails.tx.data?.parser
+      ? `transactionRow.parser.${txDetails.tx.data?.parser}.${i18n}`
+      : 'transactionRow.unknown',
+  )
+
+  const asset = useAppSelector(s => selectAssetById(s, (txDetails.tx.data as any).assetId! ?? ''))
```
Comment (0xdef1cafe) [inline src/components/TransactionHistoryRows/TransactionContract.tsx:44]: draft as i got a question about `TxMetadata` type in lib - at runtime it seems like we have `txDetails.tx.data.assetId` but not at compile time

Comment (0xdef1cafe) [reply src/components/TransactionHistoryRows/TransactionContract.tsx:44]: https://github.com/shapeshift/lib/blob/main/packages/chain-adapters/src/types.ts#L112 may need `assetId` added

Comment (0xdef1cafe) [reply src/components/TransactionHistoryRows/TransactionContract.tsx:44]: https://github.com/shapeshift/web/blob/develop/src/hooks/useTxDetails/useTxDetails.ts#L157 

this is a better place to figure the symbol

Comment (gomesalexandre) [reply src/components/TransactionHistoryRows/TransactionContract.tsx:44]: We will need to use this already exposed type from unchained-client:

https://github.com/shapeshift/lib/blob/2b89bc53b99526361200c2b61febbe6a872182a7/packages/unchained-client/src/types.ts#L77-L89

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2352** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2352" target="_blank">Graphite</a> 
    * **PR #2257** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2257" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2257?utm_source=stack-comment).

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
She works, though some package mess to fix up.

Screen Shot 2022-08-08 at 5 02 28 pm

**Review by 0xApotheosis — APPROVED** [review]
I addressed my own comments.

File: src/components/TransactionHistoryRows/TransactionContract.tsx [hunk 2]
```diff
@@ -24,24 +27,33 @@ export const TransactionContract = ({
...
   const isFirstAssetOutgoing = interactsWithWithdrawMethod && isSend
 
+  const titlePrefix = translate(
+    txDetails.tx.data?.parser
+      ? `transactionRow.parser.${txDetails.tx.data?.parser}.${i18n}`
+      : 'transactionRow.unknown',
+  )
+
+  const asset = useAppSelector(state =>
+    selectAssetById(state, isTokenMetadata(txDetails.tx.data) ? txDetails.tx.data.assetId! : ''),
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/components/TransactionHistoryRows/TransactionContract.tsx:45]: That would probably be some cached packages on your environment, it's typed properly for me

Before running `rm -rf node_modules/.cache` and the usual cleanup:

image

After cleanup and reinstall:

image

**Pull request merged**